### DISCUSSION TITLE
READ FOREIGN TABLES with curl

### DISCUSSION THREAD

**Discussion by himeshp** [supabase/supabase#1974] [Questions] [answered]
v1/newsCategory?select=some_column,other_table(foreign_key)
can anyone share a sample query how can I fetch data from foreign tables, I am seem to be struggling with it.

schema attached
<img width="1031" alt="Screenshot 2021-06-15 at 11 46 26 AM" src="https://user-images.githubusercontent.com/5878422/122002449-6157cb00-cdcf-11eb-9a50-0d3d0694ab23.png">

thanks in advance

**Comment by himeshp** [comment, marked as answer]
categoryId is a foreign relationship from the category table

**Reply by easy-street** [reply]
example if the `category` table has a column named  `foo`:
```
curl --request GET \
  --url 'https://<supabase-project-id>.supabase.co/rest/v1/newsCategory?select=title,category(foo)' \
  --header 'apikey: <supabase-anon-key>'
```

example response:
```
[
  {
    "title": "News",
    "category": {
      "foo": "bar"
    }
  }
]
```

**Reply by himeshp** [reply]
here is the complete schema,
newslist table
<img width="1031" alt="Screenshot 2021-06-15 at 11 46 26 AM" src="https://user-images.githubusercontent.com/5878422/122008786-0c1fb780-cdd7-11eb-8189-eb52cc420b42.png">

category table
<img width="683" alt="Screenshot 2021-06-15 at 12 40 51 PM" src="https://user-images.githubusercontent.com/5878422/122008735-00cc8c00-cdd7-11eb-883f-b0a5a2141b52.png">

I want to fetch the contents of the category table while calling the news list.
category id column in news list is foreign key relationship from the category table linked with id.

newsList?select=title,category(id) 

above, seems to give below error

`{
  "hint": "By following the 'details' key, disambiguate the request by changing the url to /origin?select=relationship(*) or /origin?select=target!relationship(*)",
  "details": [
    {
      "relationship": "newsList_categoryId_fkey[categoryId][id]",
      "origin": "public.newsList",
      "cardinality": "m2o",
      "target": "public.category"
    },
    {
      "relationship": "public.newsCategory[newsCategory_newsId_fkey][newsCategory_categoryId_fkey]",
      "origin": "public.newsList",
      "cardinality": "m2m",
      "target": "public.category"
    }
  ],
  "message": "More than one relationship was found for newsList and category"
}`

not sure, if I am doing something wrong here.

**Reply by easy-street** [reply]
so, in this case it's necessary to disambiguate the request. more info in the [PostgREST Embedding Disambiguation docs](https://postgrest.org/en/v7.0.0/api.html#embedding-disambiguation), as well as [previous discussions](https://github.com/supabase/supabase/discussions?discussions_q=disambiguation+).

sometimes this requires a bit of experimenting; in this case a starting point would be [hint disambiguation](https://postgrest.org/en/v7.0.0/api.html#hint-disambiguation) with the foreign key:

```
curl --request GET \
  --url 'https://<supabase-project-id>.supabase.co/rest/v1/newsList?select=title,category!newsList_categoryId_fkey(title)' \
  --header 'apikey: <supabase-anon-key>'
```

**Reply by himeshp** [reply]
Thanks for this, it works, exactly what I was looking for.
I am new to this, thanks for the help and being patient.

Regards
